### ISSUE TITLE
metric not returning: nv_inference_pending_request_count

### ISSUE THREAD

**Issue by maafk** [triton-inference-server/server#6341]
**Description**
I have a running Triton server processing async inference requests.

When I hit the metrics endpoint, I don't receive the `nv_inference_pending_request_count` in the output ([ref](https://github.com/triton-inference-server/server/blob/2ccc3cedcb6dc1712866b8f9818c819d7eaf1a00/docs/user_guide/metrics.md#counts))

**Triton Information**
`nvcr.io/nvidia/tritonserver:23.08-py3`

Are you using the Triton container or did you build it yourself?

Built the triton container using 

```Dockerfile
FROM nvcr.io/nvidia/tritonserver:23.08-py3
```

**To Reproduce**

Deployed [Stable diffusion xl 0.9](https://huggingface.co/stabilityai/stable-diffusion-xl-base-0.9).

config.pbtxt

```
name: "stable-diffusion"
backend: "python"

input [
  {
    name: "INPUT"
    data_type: TYPE_STRING
    dims: [-1]
  }
]
output [
  {
    name: "ARRAY"
    data_type: TYPE_FP32
    dims: [1024, 1024, 3]
  }
]
parameters: 

{
    key: "EXECUTION_ENV_PATH",
    value: {string_value: "$$TRITON_MODEL_DIRECTORY/environment.tar.gz"}
}
```

model.py

```python
import json
import logging
import os

import numpy as np
import torch
import triton_python_backend_utils as pb_utils
from diffusers import DiffusionPipeline

startpath = os.path.dirname(os.path.abspath(__file__))


class TritonPythonModel:
    def initialize(self, args):
        self.model_config = model_config = json.loads(args["model_config"])
        self.pipe = DiffusionPipeline.from_pretrained(
            startpath, torch_dtype=torch.float16, use_safetensors=True, variant="fp16"
        )
        self.pipe.to("cuda")

        self.pipe.unet = torch.compile(
            self.pipe.unet, mode="reduce-overhead", fullgraph=True
        )

    def execute(self, requests):
        responses = []

        for request in requests:
            input_sync = pb_utils.get_input_tensor_by_name(request, "INPUT")
            prompt = input_sync.as_numpy()[0].decode()

            output = self.pipe(prompt=prompt).images[0]

            array_output_config = pb_utils.get_output_config_by_name(
                self.model_config, "ARRAY"
            )

            output = np.array(output, dtype=np.float32)
            output = output.reshape((1024, 1024, 3))

            output_dtype = pb_utils.triton_string_to_numpy(
                array_output_config["data_type"]
            )
            out_tensor = pb_utils.Tensor("ARRAY", output.astype(output_dtype))

            inference_response = pb_utils.InferenceResponse(output_tensors=[out_tensor])
            responses.append(inference_response)

        return responses

    def finalize(self):
        logging.info("Cleaning up...")
```

command to run the triton service

```zsh
docker run --gpus=1 --shm-size=8gb \
  --rm \
  -p8000:8000 -p8001:8001 -p8002:8002 \
  tritonserver \
  --model-repository=s3://my-bucket/path/to/model_repository --log-verbose=3
```

Then I run a bunch of async inference requests that _succeed_ eventually

```python
import tritonclient.grpc as triton
import numpy as np

client = triton.InferenceServerClient(
    url=f"localhost:8001",
)

prompts = [
    b"A rabbit howling at the moon",
    b"A panda bear eating birthday cake",
    b"Two parrots playing chess",
    b"A Ford F-150 truck being pulled by a unicorn",
    b"A well dressed Englishman standing in a pool of water",
    b"A dog sitting at a desk, pretending to work on a computer.",
]

def async_callback(result, error):
    if error:
        print(error)
    else:
        print("done")

for prompt in prompts:
    input_data = np.array([prompt])
    input_tensor = [triton.InferInput("INPUT", input_data.shape, "BYTES")]
    input_tensor[0].set_data_from_numpy(input_data)

    response = client.async_infer(
        "stable-diffusion", input_tensor, callback=async_callback
    )
```

**Expected behavior**

When I query the server metrics, I expect to see the `nv_inference_pending_request_count`, but it doesn't show

```zsh
curl localhost:8002/metrics 
# HELP nv_inference_request_success Number of successful inference requests, all batch sizes
# TYPE nv_inference_request_success counter
nv_inference_request_success{model="stable-diffusion",version="1"} 72
# HELP nv_inference_request_failure Number of failed inference requests, all batch sizes
# TYPE nv_inference_request_failure counter
nv_inference_request_failure{model="stable-diffusion",version="1"} 0
# HELP nv_inference_count Number of inferences performed (does not include cached requests)
# TYPE nv_inference_count counter
nv_inference_count{model="stable-diffusion",version="1"} 72
# HELP nv_inference_exec_count Number of model executions performed (does not include cached requests)
# TYPE nv_inference_exec_count counter
nv_inference_exec_count{model="stable-diffusion",version="1"} 72
# HELP nv_inference_request_duration_us Cumulative inference request duration in microseconds (includes cached requests)
# TYPE nv_inference_request_duration_us counter
nv_inference_request_duration_us{model="stable-diffusion",version="1"} 45092321335
# HELP nv_inference_queue_duration_us Cumulative inference queuing duration in microseconds (includes cached requests)
# TYPE nv_inference_queue_duration_us counter
nv_inference_queue_duration_us{model="stable-diffusion",version="1"} 43940359101
# HELP nv_inference_compute_input_duration_us Cumulative compute input duration in microseconds (does not include cached requests)
# TYPE nv_inference_compute_input_duration_us counter
nv_inference_compute_input_duration_us{model="stable-diffusion",version="1"} 2020
# HELP nv_inference_compute_infer_duration_us Cumulative compute inference duration in microseconds (does not include cached requests)
# TYPE nv_inference_compute_infer_duration_us counter
nv_inference_compute_infer_duration_us{model="stable-diffusion",version="1"} 1151632767
# HELP nv_inference_compute_output_duration_us Cumulative inference compute output duration in microseconds (does not include cached requests)
# TYPE nv_inference_compute_output_duration_us counter
nv_inference_compute_output_duration_us{model="stable-diffusion",version="1"} 326217
# HELP nv_energy_consumption GPU energy consumption in joules since the Triton Server started
# TYPE nv_energy_consumption counter
nv_energy_consumption{gpu_uuid="GPU-9ad0ded3-efbf-491e-928b-0031070d270b"} 309409.2330000002
# HELP nv_gpu_utilization GPU utilization rate [0.0 - 1.0)
# TYPE nv_gpu_utilization gauge
nv_gpu_utilization{gpu_uuid="GPU-9ad0ded3-efbf-491e-928b-0031070d270b"} 0.99
# HELP nv_gpu_memory_total_bytes GPU total memory, in bytes
# TYPE nv_gpu_memory_total_bytes gauge
nv_gpu_memory_total_bytes{gpu_uuid="GPU-9ad0ded3-efbf-491e-928b-0031070d270b"} 24146608128
# HELP nv_gpu_memory_used_bytes GPU used memory, in bytes
# TYPE nv_gpu_memory_used_bytes gauge
nv_gpu_memory_used_bytes{gpu_uuid="GPU-9ad0ded3-efbf-491e-928b-0031070d270b"} 15667822592
# HELP nv_gpu_power_usage GPU power usage in watts
# TYPE nv_gpu_power_usage gauge
nv_gpu_power_usage{gpu_uuid="GPU-9ad0ded3-efbf-491e-928b-0031070d270b"} 263.813
# HELP nv_gpu_power_limit GPU power management limit in watts
# TYPE nv_gpu_power_limit gauge
nv_gpu_power_limit{gpu_uuid="GPU-9ad0ded3-efbf-491e-928b-0031070d270b"} 300
# HELP nv_cpu_utilization CPU utilization rate [0.0 - 1.0]
# TYPE nv_cpu_utilization gauge
nv_cpu_utilization 0.12625
# HELP nv_cpu_memory_total_bytes CPU total memory (RAM), in bytes
# TYPE nv_cpu_memory_total_bytes gauge
nv_cpu_memory_total_bytes 33345327104
# HELP nv_cpu_memory_used_bytes CPU used memory (RAM), in bytes
# TYPE nv_cpu_memory_used_bytes gauge
nv_cpu_memory_used_bytes 5778485248
```

Is there additional configurations I need to ensure I can get the `nv_inference_pending_request_count` metric back?
